src/codegen/function.rs

```diff
@@ -456,3 +459,41 @@ pub fn body_chunk_futures(
 
     Ok(body)
 }
+
+pub(super) fn generate_static_to_str(
+    w: &mut dyn Write,
+    env: &Env,
+    function: &analysis::functions::Info,
+    scope_version: Option<Version>,
+) -> Result<()> {
+    writeln!(w)?;
+    let version = Version::if_stricter_than(function.version, scope_version);
+    version_condition(w, env, version, false, 1)?;
+
+    let visibility = match function.visibility {
+        Visibility::Public => "pub ",
+        _ => "",
+    };
+
+    writeln!(
+        w,
+        "\
+\t{visibility}fn {rust_fn_name}(self) -> &'static str {{
```

Comment:
The `\t` everywhere are making the reading super complicated... We'll need to improve that later on.

Reply:
@GuillaumeGomez I've simply adopted the standards used elsewhere in `gir`, but this is one of the larger predefined codeblocks making it more pronounced. This is why I've been advocating for `quote`, for which I have a PR in the works (submitted after this one) that replaces most of the code blocks in this PR.

Reply:
I think it's fine as is. It's bad but consistent with code elsewhere.

Once this is in we should incrementally start moving to a better code generation mechanism.

Reply:
@sdroege Yeah I think the only consistent, _readable_ solution is quote :grimacing: 

I'll rebase and refactor most of these blocks when this PR is in :)

Reply:
`quote` didn't help much iirc. I think the best way is just plain string formatting but in much bigger steps instead of all the small ones we currently have.

Reply:
I made quite a case for it in https://github.com/gtk-rs/gir/pull/992#discussion_r524124393, and it's only getting better than that :)